.github/workflows/test-setup.yaml

```diff
@@ -0,0 +1,29 @@
+name: test-setup
+
+on:
+  # The test requires idtoken permissions, which are not available on
+  # pull_request. This means the workflow config will be taken from HEAD,
+  # so we checkout the PR merge commit explicitly below.
```

Comment:
This also means this test won't run until we merge it. Which sucks. If it fails I can iterate on a fix, but it'll be slow and annoying because the workflow comes from HEAD. 😒